morph/react/block-add-data.js

```diff
@@ -38,29 +34,98 @@ function getDataVariableName(data) {
       .map(toSnakeCase)
       .join('_')
   )}`
+  if (state.usedDataNames[name]) {
+    name = `${name}${state.usedDataNames[name]}`
+    state.usedDataNames[name] += 1
+  } else {
+    state.usedDataNames[name] = 1
+  }
+  return name
 }
 
-function maybeDataContext(dataDefinition, data) {
+function maybeDataContext(dataDefinition, state) {
   if (dataDefinition.context === null) return
 
-  data.push(`context: '${dataDefinition.context}',`)
+  state.dataBlocks.push(`context: '${dataDefinition.context}',`)
 }
 
-function maybeDataFormat(format, data) {
+function maybeDataFormat(format, state) {
   if (!format) return
 
   if (format.formatIn) {
-    data.push(`formatIn: '${format.formatIn}',`)
+    let importName = getFormatImportName(format.formatIn.source, state)
+    state.dataBlocks.push(`formatIn: ${importName}.${format.formatIn.value},`)
   }
 
   if (format.formatOut) {
-    data.push(`formatOut: '${format.formatOut}',`)
+    let importName = getFormatImportName(format.formatOut.source, state)
+    state.dataBlocks.push(`formatOut: ${importName}.${format.formatOut.value},`)
   }
 }
-function maybeDataValidate(validate, data) {
+
+function maybeDataValidate(validate, state) {
   if (!validate || validate.type !== 'js') return
-  data.push(`validate: '${validate.value}',`)
+  let importName = getValidateImportName(validate.source, state)
+  state.dataBlocks.push(`validate: ${importName}.${validate.value},`)
   if (validate.required) {
-    data.push('validateRequired: true,')
+    state.dataBlocks.push('validateRequired: true,')
+  }
+}
+
+function getFilePath(source) {
+  if (path.isAbsolute(source)) {
+    return source.startsWith('/') ? source.substring(1) : source
+  }
+  if (source.startsWith('.')) return source
+  return `./${source}`
+}
```

Comment:
Could a path be absolute and not start with `/`?

Reply:
I just wanted to be extra safe, probably not needed so I'll get rid of this check for now